File: LogManager.cpp
#include "LogManager.h"
#include <QProcess>

LogManager::LogManager(QObject *parent) : QObject(parent)
{
}

LogManager::~LogManager()
{
}

bool LogManager::uploadFile(const QString &localFilePath, const QString &ossPath)
{
    // 构建上传命令
    QString command = QString("ossutil cp -f %1 %2").arg(localFilePath, ossPath);

    return executeCommand(command);  // 执行命令
}

bool LogManager::downloadFile(const QString &ossPath, const QString &localFilePath)
{
    // 构建下载命令
    QString command = QString("ossutil cp -f %1 %2").arg(ossPath, localFilePath);

    return executeCommand(command);  // 执行命令
}

bool LogManager::appendFile(const QString &localFilePath, const QString &ossPath)
{
    // 构建下载命令
    QString command = QString("ossutil appendfromfile -f %1 %2").arg(localFilePath, ossPath);

    return executeCommand(command);  // 执行命令
}

bool LogManager::executeCommand(const QString &command)
{
    QProcess process;
    
    // 先拆分命令和参数
    QStringList arguments = command.split(" ");
    QString program = arguments.takeFirst();  // 第一个元素是程序名

    // 启动进程
    process.start(program, arguments);
    process.waitForFinished();  // 等待命令执行完毕

    if (process.exitStatus() == QProcess::NormalExit && process.exitCode() == 0) {
        // 获取命令输出
        QByteArray output = process.readAllStandardOutput();
        qDebug() << "Command Output:" << output;
        return true;  // 命令执行成功
    } else {
        // 获取错误输出
        QByteArray errorOutput = process.readAllStandardError();
        qDebug() << "Error Output:" << errorOutput;
        return false;  // 命令执行失败
    }
}

File: log_manager.py
import subprocess
from tkinter import END

class LogManager:
    def __init__(self):
        """ 初始化 LogManager 类 """

    def upload_file(self, local_file_path, oss_path):
        """
        上传文件到 OSS
        Args:
            local_file_path: 本地文件路径
            oss_path: OSS 目标路径
        Returns:
            bool: 上传是否成功
        """

        try:
            # 构建上传命令
            command = f"ossutil cp -f {local_file_path} {oss_path}"
            result = subprocess.run(command, shell=True, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
            # print("Upload successful")
            return True  # 上传成功

        except subprocess.CalledProcessError as e:
            return False  # 上传失败

    def download_file(self, oss_path, local_file_path):
        """
        从 OSS 下载文件
        Args:
            oss_path: OSS 文件路径
            local_file_path: 本地保存路径
        Returns:
            bool: 下载是否成功
        """

        try:
            # 构建下载命令
            command = f"ossutil cp -f {oss_path} {local_file_path}"
            result = subprocess.run(command, shell=True, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
            # print("Download successful")
            return True  # 下载成功

        except subprocess.CalledProcessError as e:
            print("Error during download:", e.stderr)
            return False  # 下载失败
